row.go

```diff
@@ -521,14 +530,26 @@ func reconstructFuncOf(columnIndex int16, node Node) (int16, reconstructFunc) {
 
 //go:noinline
 func reconstructFuncOfOptional(columnIndex int16, node Node) (int16, reconstructFunc) {
+	// What's the purpose of forcing the passed in node to Required for the Optional fields?
+	// Ah, I see it's so that we eventually reach the leaf base-case?
 	nextColumnIndex, reconstruct := reconstructFuncOf(columnIndex, Required(node))
+	// What if this is the last node/column in the row? We just incremented the
+	// nextColumnIndex meaning that if this is the last one in the list,
+	// rowLength will always be > 0 here, triggering the second if statement in
+	// the closure below, even if the optional field is empty
 	rowLength := nextColumnIndex - columnIndex
```

Comment:
I still don't understand why this is here and don't have it in the convert code yet.